internal/v1/handler_post_compose_test.go

```diff
@@ -2946,6 +2941,100 @@ func TestComposeCustomizations(t *testing.T) {
 				},
 			},
 		},
+		// aap first boot registration
+		{
+			imageBuilderRequest: v1.ComposeRequest{
+				Customizations: &v1.Customizations{},
+				Distribution:   "rhel-8",
+				ImageRequests: []v1.ImageRequest{
+					{
+						Architecture: "x86_64",
+						ImageType:    v1.ImageTypesGuestImage,
+						UploadRequest: v1.UploadRequest{
+							Type:    v1.UploadTypesAwsS3,
+							Options: uo,
+						},
+						AAPRegistration: &v1.AAPRegistration{
+							AnsibleControllerUrl:    "http://some-url.org",
+							HostConfigKey:           "some-key",
+							JobTemplateId:           38,
+							TlsCertificateAuthority: common.ToPtr("---BEGIN CERTIFICATE---\nMIIC0DCCAbigAwIBAgIUI...\n---END CERTIFICATE---"),
+						},
+					},
+				},
+			},
+			composerRequest: composer.ComposeRequest{
+				Distribution: "rhel-8.10",
+				Customizations: &composer.Customizations{
+					Files: &[]composer.File{
+						{
+							Data: common.ToPtr(fmt.Sprintf(
+								"#!/usr/bin/bash\ncurl -s -i --data %q %s",
+								fmt.Sprintf("host_config_key=%s", "some-key"),
+								fmt.Sprintf(
+									"%s/api/v2/job_templates/%v/callback/",
+									"http://some-url.org",
+									"38",
+								),
+							)),
+							Path:          "/usr/local/sbin/aap-first-boot-reg",
+							Mode:          common.ToPtr("0774"),
+							EnsureParents: common.ToPtr(true),
+						},
+						{
+							Data: common.ToPtr(`
+Description=Ansible Automations Platform first boot registration
+ConditionFileIsExecutable=/usr/local/sbin/aap-first-boot-reg
+ConditionPathExists=!/var/local/.aap-first-boot-reg-done
+Wants=network-online.target
+After=network-online.target
+After=osbuild-first-boot.service
+
+[Service]
+Type=oneshot
+ExecStart=/usr/local/sbin/aap-first-boot-reg
+ExecStartPost=/usr/bin/touch /var/local/.aap-first-boot-reg-done
+RemainAfterExit=yes
+
+[Install]
+WantedBy=basic.target
+WantedBy=multi-user.target
+WantedBy=graphical.target
+`),
```

Comment:
Can you drop both basic and graphical? This is a leftover I am going to fix in the UI.